src/OnboardingSPA/pages/Steps/DesignColors/utils.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * Helper Function for state in global styles to local mapping
+ *
+ * @param {Array} selectedColorPalette - Array of Map structure similar to the one in Global Styles
+ *                                     e.g. [{color: '#ffffff', name: 'Base', slug: 'base'}, ...]
+ *                                     return -> {base: "#ffffff", contrast: "#404040", ... }
+ */
+export function storeToState( selectedColorPalette ) {
+	if ( selectedColorPalette ) {
+		const selectedColorsLocalTemp = {};
+		selectedColorPalette?.forEach( ( color ) => {
+			selectedColorsLocalTemp[ color.slug ] = color.color;
+		} );
+		return selectedColorsLocalTemp;
+	}
+}
+
+/**
+ * Converts the user selected value into a suitable valid global styles array value
+ *
+ * @param {Object} selectedColorsLocalTemp - Color type mapped to the color
+ *                                         e.g. {base: "#ffffff", contrast: "#404040", ... }
+ * @param {string} colorStyle              - Selected Color slug
+ *                                         e.g. [{color: '#ffffff', name: 'Base', slug: 'base'}, ...]
+ */
+export function stateToStore( selectedColorsLocalTemp, colorStyle ) {
```

Comment:
```suggestion
export function stateToStore( selectedColorsInState, colorStyle ) {
```
Also do we actually need `colorStyle`, I can see it is part of the `if` but it is not being used anywhere.